File: chapter_03/3_5.py
a_list = [8, 12, 16]

count = 10000

# 配列の要素がすべて偶数か調べる
is_all_even = True
for a in a_list:
    if a%2 != 0:
        is_all_even = False

if is_all_even: # 配列の要素がすべて偶数の場合
    for a in a_list:
        temp_count = 0
        while(a%2 == 0):
            a = a / 2
            temp_count += 1
        if temp_count < count:
            count = temp_count

    print(f"最小は{count}") # 答えは２（要素１２の場合）

File: chapter_03/c++/3_5.cpp
#include<string>
#include<vector>
#include<iostream>
#include<climits>

using namespace std;

int main(){
    vector<int> list_a = {8, 12, 16};

    int count = INT_MAX;

    // 全ての要素が偶数か判定
    bool are_all_even = true;
    for (int i=0; i<list_a.size(); i++){
        if (list_a[i] % 2 != 0){
            are_all_even = false;
            break;
        }
    }

    // 何回割り算が行えるかの最小値を計算
    for (int i=0; i<list_a.size(); i++){
        int a = list_a[i];
        int temp_count = 0;
        while (true){
            if (a % 2 == 0){
                a = a / 2;
                temp_count ++;
            }
            else break;
        }

        // temp_countがcountより小さければcountを更新
        if (temp_count < count){
            count = temp_count;
        }
    }

    cout << "minimum count : " << count << endl;
}